crates/networking/p2p/peer_channels.rs

```diff
@@ -447,4 +447,68 @@ impl PeerChannels {
             })
             .flatten()
     }
+
+    /// Requests a single storage range for an accouns given its hashed address and storage root, and the root of its state trie
+    /// This is a simplified version of `request_storage_range` meant to be used for large tries that require their own single requests
+    /// account_hashes & storage_roots must have the same length
+    /// storage_root must not be an empty trie hash, we will treat empty ranges as invalid responses
+    /// Returns true if the account's storage was not completely fetched by the request
+    /// Returns the list of hashed storage keys and values for the account's storage or None if:
+    /// - There are no available peers (the node just started up or was rejected by all other nodes)
+    /// - The response timed out
+    /// - The response was empty or not valid
+    pub async fn request_storage_range(
+        &self,
+        state_root: H256,
+        storage_root: H256,
+        account_hash: H256,
+        start: H256,
+    ) -> Option<(Vec<H256>, Vec<U256>, bool)> {
+        let request_id = rand::random();
+        let request = RLPxMessage::GetStorageRanges(GetStorageRanges {
+            id: request_id,
+            root_hash: state_root,
+            account_hashes: vec![account_hash],
+            starting_hash: start,
+            limit_hash: HASH_MAX,
+            response_bytes: MAX_RESPONSE_BYTES,
+        });
+        let mut receiver = self.receiver.lock().await;
+        self.sender.send(request).await.ok()?;
+        let (mut slots, proof) = tokio::time::timeout(PEER_REPLY_TIMOUT, async move {
+            loop {
+                match receiver.recv().await {
```

Comment:
I think you mentioned this in the daily, but it totally makes sense to abstract the common logic of communication via channel. Maybe that logic should live on this file and the we should move the specific functions like `request_storage_range` to another file. If I would want to search for how storage ranges are requested, I would never think of going to a file named `peer_channels`. Also maybe we could create a `snap` folder that contains some of this code. Not saying you should do it on this PR, but eventually.

Reply:
This is quite generic code that creates a request, sends it, and expects a response, I don't think it really merits its own snap module, the logic to fulfill the snap request is already on the snap module. What I want so abstract is peer selection and request retry so we don't need to have it intertwined with the sync logic.